internal/store/store.go

```diff
@@ -512,7 +513,7 @@ func (s Store) ListIngressClassParametersV1Alpha1() []*kongv1alpha1.IngressClass
 	for _, item := range s.stores.IngressClassParametersV1alpha1.List() {
 		classParam, ok := item.(*kongv1alpha1.IngressClassParameters)
 		if !ok {
-			s.logger.Warnf("listIngressClassParametersV1alpha1: dropping object of unexpected type: %#v", item)
+			s.logger.Error(nil, "listIngressClassParametersV1alpha1: dropping object of unexpected type", "type", fmt.Sprintf("%#v", item))
```

Comment:
```suggestion
			s.logger.Error(nil, "listIngressClassParametersV1alpha1: dropping object of unexpected type", "type", fmt.Sprintf("%T", item))
```

Reply:
I believe this doesn't happen too often but I'd change all those calls to what's in the suggestion as that's the actual "type" of the objects, additionally we could add a field with the `%#v` of the `item`.

I can see it's done similarly in other log site so it might be done in a separate PR to prevent this one from getting even bigger.